tests/unit/dataset/test_dataset.py

```diff
@@ -394,57 +395,78 @@ def test__dataframe__data_type_field_not_exist() -> None:
         assert DATA_TYPE_FIELD not in row[column_name]
 
 
-def test__infer_id_fields() -> None:
-    assert _infer_id_fields(
-        pd.DataFrame(
-            dict(
-                locator=["s3://test.pdf", "https://test.png", "/home/test.mp4", "/tmp/test.pcd"],
+@pytest.mark.parametrize(
+    "input_df, expected",
+    [
+        (
+            pd.DataFrame(
+                dict(
+                    id=[1, 2, 3, 4],
+                    locator=["s3://test.pdf", "https://test.png", "/home/test.mp4", "/tmp/test.pcd"],
+                ),
             ),
+            ["id"],
         ),
-    ) == ["locator"]
-    assert _infer_id_fields(
-        pd.DataFrame(
-            dict(
-                locator=["s3://test.pdf", "https://test.png", "/home/test.mp4", "/tmp/test.pcd"],
-                text=["a", "b", "c", "d"],
+        (
+            pd.DataFrame(
+                dict(
+                    locator=["s3://test.pdf", "https://test.png", "/home/test.mp4", "/tmp/test.pcd"],
+                ),
             ),
+            ["locator"],
         ),
-    ) == ["locator"]
-    assert _infer_id_fields(
-        pd.DataFrame(
-            dict(
-                text=["a", "b", "c", "d"],
+        (
+            pd.DataFrame(
+                dict(
+                    locator=["s3://test.pdf", "https://test.png", "/home/test.mp4", "/tmp/test.pcd"],
+                    text=["a", "b", "c", "d"],
+                ),
             ),
+            ["locator"],
         ),
```

Comment:
nit: do we need both tests? I think the test with locator and text has us covered since we are just checking priority order